spec/services/solidus_square/webhooks/handlers/payment_updated_spec.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+RSpec.describe SolidusSquare::Webhooks::Handlers::PaymentUpdated do
+  # subject(:handler) { described_class.new("test") }
+
+  # let(:service) { instance_double(SolidusSquare::PaymentSyncService) }
+  # still need to find a way how to make this work...
+
+  # before do
+  #   allow(SolidusSquare::PaymentSyncService).to receive(:call)
+  # end
+  it { expect(2 + 2).to eq(4) }
+  # it { expect(handler.call).to have_receive(SolidusSquare::PaymentSyncService.call) }
```

Comment:
You are pretty close to the solution, it could be something like:

```
subject(:handler) { described_class.new("test") }

  before do
    allow(SolidusSquare::PaymentSyncService).to receive(:call)
  end

  it 'calls SolidusSquare::PaymentSyncService service object' do
    handler.call

    expect(SolidusSquare::PaymentSyncService).to have_received(:call).with('test')
  end
  ```
  
  TO READ https://relishapp.com/rspec/rspec-mocks/v/3-10/docs/basics/allowing-messages